packages/quantic/force-app/main/default/lwc/quanticNumericFacet/__tests__/quanticNumericFacet.test.js

```diff
@@ -119,85 +175,212 @@ function mockSuccessfulHeadlessInitialization() {
   };
 }
 
-function cleanup() {
-  // The jsdom instance is shared across test cases in a single file so reset the DOM
-  while (document.body.firstChild) {
-    document.body.removeChild(document.body.firstChild);
-  }
-  jest.clearAllMocks();
-  isInitialized = false;
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticNumericFacet) {
+      element.setInitializationError();
+    }
+  };
+}
+
+function setupEventDispatchTest(eventName) {
+  const handler = (event) => {
+    functionsMocks.eventHandler(event?.detail);
+    document.removeEventListener(eventName, handler);
+  };
+  document.addEventListener(eventName, handler);
+}
+
+async function submitFacetSearchForm(element) {
+  const form = element.shadowRoot.querySelector(selectors.searchForm);
+  form.dispatchEvent(
+    new CustomEvent('submit', {bubbles: true, composed: true})
+  );
+  await flushPromises();
+}
+
+async function mockInputReportValidity(element) {
```

Comment:
similar to the bueno comment, should this also be moved to utils file?

Reply:
No really as this rarely used in our components.